### PR TITLE
Add endpoint to get category with processingType

### PR DESCRIPTION
**Pull request by Filemon279**

### CODE REVIEW

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala [hunk 1]
```diff
@@ -37,6 +40,18 @@ class DefinitionResources(modelDataProvider: ProcessingTypeDataProvider[ModelDat
           modelDataProvider.mapValues(_.processDefinition.services.mapValuesNow(UIProcessObjectsFactory.createUIObjectDefinition(_, processCategoryService))).all
         }
       }
+    } ~ path("processDefinitionData" / "categoriesWithProcessingType") {
+      get {
+        complete {
+          val categories = user match {
+            case user: CommonUser => processCategoryService.getAllCategories.filter(user.can(_, Read))
```
Comment (gadomsky) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala:47]: why not `processCategoryService.getUserCategories` ?

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala [hunk 2]
```diff
@@ -37,6 +40,18 @@ class DefinitionResources(modelDataProvider: ProcessingTypeDataProvider[ModelDat
...
+            case _: AdminUser => processCategoryService.getAllCategories
+          }
+          categories
+            .map(category => category -> processCategoryService.getTypeForCategory(category))
+            .toMap[Category, Option[ProcessingType]]
```
Comment (gadomsky) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala:52]: is it possible, that there is no ProcessingType (None) for a category?

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala (hunk 1 above)
Comment (Filemon279) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala]: Didn't noticed that so I reinvented the wheel .. Thanks, changed.

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala (hunk 2 above)
Comment (Filemon279) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala:48]: API defines `getTypeForCategory` with `Option[ProcessingType]` so I didn't try to go deep into it. As I look at it now I think that in my case it always have processingType but I can see cases when this `getTypeForCategory` can return None. 

We could drop Option but it wouldn't make any difference. WDYT?

Comment (gadomsky) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala:48]: ok lets leave it

**Review by gadomsky — APPROVED** [review]

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala [hunk 3]
```diff
@@ -37,6 +40,14 @@ class DefinitionResources(modelDataProvider: ProcessingTypeDataProvider[ModelDat
           modelDataProvider.mapValues(_.processDefinition.services.mapValuesNow(UIProcessObjectsFactory.createUIObjectDefinition(_, processCategoryService))).all
         }
       }
+    } ~ path("processDefinitionData" / "categoriesWithProcessingType") {
+      get {
+        complete {
+            processCategoryService.getUserCategories(user)
```
Comment (lciolecki) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala:46]: Hmm, maybe better way is to create one method for userCategoriesType?

Comment (Filemon279) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala]: No problem'o - I've added `getUserCategoriesWithType` method and it helped me to drop `Option` from return type without any handlers. :+1:

File: designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala [hunk 4]
```diff
@@ -37,6 +40,12 @@ class DefinitionResources(modelDataProvider: ProcessingTypeDataProvider[ModelDat
           modelDataProvider.mapValues(_.processDefinition.services.mapValuesNow(UIProcessObjectsFactory.createUIObjectDefinition(_, processCategoryService))).all
         }
       }
+    } ~ path("processDefinitionData" / "categoriesWithProcessingType") {
```
Comment (lciolecki) [inline designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala:43]: One thing more. This is not processDefinitionData, but just settings / config?

Comment (Filemon279) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala]: I've moved it next to the config endpoint.

Comment (lciolecki) [reply designer/server/src/main/scala/pl/touk/nussknacker/ui/api/DefinitionResources.scala]: Ofc, It will stay here forever :)

**Pull request merged**